src/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceInternal.cs

```diff
@@ -56,7 +56,7 @@ internal static string AppName
             {
                 if (s_appName == null)
                 {
-                    s_appName = Assembly.GetEntryAssembly().GetName().Name;
+                    s_appName = Assembly.GetEntryAssembly()?.GetName().Name ?? string.Empty;
```

Comment:
Getname can never return null?

Reply:
Right. And if it could, there are a bunch of other places that access it without null checks that would need to be updated.